port/cpl_http.h

```diff
@@ -179,8 +179,47 @@ CPL_C_END
 void CPL_DLL *CPLHTTPSetOptions(void *pcurl, const char *pszURL,
                                 const char *const *papszOptions);
 char **CPLHTTPGetOptionsFromEnv(const char *pszFilename);
-double CPLHTTPGetNewRetryDelay(int response_code, double dfOldDelay,
-                               const char *pszErrBuf, const char *pszCurlError);
+
+/** Stores HTTP retry parameters */
+struct CPLHTTPRetryParameters
+{
+    int nMaxRetry = CPL_HTTP_MAX_RETRY;
+    double dfInitialDelay = CPL_HTTP_RETRY_DELAY;
+    std::string osRetryCodes{};
+
+    CPLHTTPRetryParameters() = default;
+    explicit CPLHTTPRetryParameters(const CPLStringList &aosHTTPOptions);
+};
+
+/** HTTP retry context */
+class CPLHTTPRetryContext
+{
+  public:
+    explicit CPLHTTPRetryContext(const CPLHTTPRetryParameters &oParams);
+
+    bool CanRetry(int response_code, const char *pszErrBuf,
+                  const char *pszCurlError);
+    bool CanRetry();
+
+    /** Returns the delay to apply. Only valid after a successful call to CanRetry() */
```

Comment:
Reading this note makes me wonder if it is worth adding a debug message to prevent misuse (calling this without calling CanRetry).

Reply:
done (testing m_nRetryCount == 0)